docs/sources/reference/builder.md

```diff
@@ -650,6 +672,28 @@ The copy obeys the following rules:
 - If `<dest>` doesn't exist, it is created along with all missing directories
   in its path.
 
+### Set ownership of content added by COPY
+
+By default, all new files and directories added by the `COPY` instruction are
+created with a UID and GID of 0. To copy files with a different owner, specify
+a user or uid using this form:
+
+    COPY --user=<uid_or_username> <src>... <dest>
+
+All files and directories copied to the container will be owned by the
+specified `<uid_or_username>`. Use of the `--user` flag will only affect the current
+`COPY` command.
+
+For example, the following will copy all files and folders from the `/proj/app`
+directory on the host to `/app` in the container. All files copied to the
+container will be owned by user "foobar":
+
+    COPY --user=foobar /projects/app /app
+
+> **Note**:
+> The specified user or uid must *exist* inside the container before running
+> the `COPY` command. Docker does not automatically create the user.
```

Comment:
Actually, a `uid` (numeric) doesn't have to exist.
(not sure what happens when an unknown `username` is used)